test/rules.jl

```diff
@@ -0,0 +1,74 @@
+using Enzyme
+using Enzyme: EnzymeRules
+using Test
+
+@testset "Custom Forward Rules" begin
+
+    rule_f(x) = x^2
+
+    function rule_fip(x)
+       x[1] *= x[1]
+       return nothing
+    end
+
+    function EnzymeRules.forward(::Type{Tuple{typeof(rule_f), Float64}}, RT, Args)
+        @assert Args[1] <: Const
```

Comment:
okay we still need to pass width explicitly as this could take no args and allocate